File: bindings/nostr-ffi/bindings-python/examples/event_builder.py
from nostr_protocol import Keys, PublicKey, EventBuilder, Event, Tag

keys = Keys.generate()

# Build a text note
event = EventBuilder.new_text_note("New note from Rust Nostr python bindings", []).to_event(keys)
print(event.as_json())

# Build a DM
receiver_pk = PublicKey.from_bech32("npub14f8usejl26twx0dhuxjh9cas7keav9vr0v8nvtwtrjqx3vycc76qqh9nsy")
event = EventBuilder.new_encrypted_direct_msg(keys, receiver_pk, "New note from Rust Nostr python bindings", None).to_event(keys)
print(event.as_json())

# Build a custom event
kind = 1234
content = "My custom content"
tags = []
builder = EventBuilder(kind, content, tags)

# Normal
event = builder.to_event(keys)
print(f"Event: {event.as_json()}")

# POW
event = builder.to_pow_event(keys, 20)
print(f"POW event: {event.as_json()}")

# Unsigned
event = builder.to_unsigned_event(keys.public_key())
print(f"Event: {event.as_json()}")

File: bindings/nostr-sdk-js/examples/event-builder.js
const { Keys, EventBuilder, Kind, Tag, loadWasmSync, Timestamp } = require("../");

function main() {
    loadWasmSync();
    
    // Generate new random keys
    let keys = Keys.generate();

    let kind = new Kind(1);
    let event = new EventBuilder(kind, "Testing nostr JS bindings", [Tag.parse(["p", "d0a59cd44b6051708e9d437aa01f86451378a130ea7ba38ad43eae0bd0e0c4ce"])]).toEvent(keys);
    console.log(event.asJson()); // Print event as JSON
    console.log(event.tags[0].toVec()); // Print first tag

    // Custom created at
    let customTimestamp = Timestamp.fromSecs(12345);
    let e = EventBuilder.textNote("Event with custom timestamp", []).customCreatedAt(customTimestamp).toEvent(keys);
    console.log(e.asJson());
}

main();